controllers/storagecluster/noobaa_system_reconciler.go

```diff
@@ -206,9 +207,9 @@ func (r *StorageClusterReconciler) setNooBaaDesiredState(nb *nbv1.NooBaa, sc *oc
 					// Secret is created by UI in "openshift-storage" namespace
 					nb.Spec.Security.KeyManagementService.TokenSecretName = KMSTokenSecretName
 				}
-			} else if kmsConfig.Data["KMS_PROVIDER"] == IbmKeyProtectKMSProvider {
+			} else if kmsProviderName == IbmKeyProtectKMSProvider || kmsProviderName == ThalesKMSProvider {
```

Comment:
[IbmKeyProtectKMSProvider, ThalesKMSProvider].includes(kmsProviderName), its is same but will reduce number of OR condition